runtime/src/main/java/com/flipkart/flux/guice/module/HibernateModule.java

```diff
@@ -55,4 +65,49 @@ protected void configure() {
         bindInterceptor(Matchers.any(), Matchers.annotatedWith(Transactional.class), transactionInterceptor);
     }
 
+    /**
+     * Adds annotated classes and custom types to passed Hibernate configuration.
+     */
+    private void addAnnotatedClassesAndTypes(Configuration configuration) {
+        //register hibernate custom types
+        configuration.registerTypeOverride(new BlobType(), new String[]{"BlobType"});
+        configuration.registerTypeOverride(new StoreFQNType(), new String[]{"StoreFQNOnly"});
+
+        //add annotated classes to configuration
+        configuration.addAnnotatedClass(AuditRecord.class);
+        configuration.addAnnotatedClass(Event.class);
+        configuration.addAnnotatedClass(State.class);
+        configuration.addAnnotatedClass(StateMachine.class);
+    }
+
+    /**
+     * Creates hibernate configuration from the configuration yaml properties.
+     * Since the yaml properties are already flattened in input param <code>yamlConfiguration</code>
+     * the method loops over them to selectively pick Hibernate specific properties.
+     */
+    @Provides
+    @Singleton
+    public Configuration getConfiguration(YamlConfiguration yamlConfiguration) {
+        Configuration configuration = new Configuration();
+        addAnnotatedClassesAndTypes(configuration);
+        Iterator<String> propertyKeys = yamlConfiguration.getKeys();
+        while (propertyKeys.hasNext()) {
```

Comment:
Instead of checking startsWith you can use method the below method.
Configuration configuration = yamlConfiguration.subset(key); //key is Hibernate in this case
This returns all config properties which starts with "Hibernate."(including dot), basically which are under element Hibernate